google/cloud/internal/oauth2_google_credentials_test.cc

```diff
@@ -108,9 +139,46 @@ TEST_F(GoogleCredentialsTest, LoadValidAuthorizedUserCredentialsViaGcloudFile) {
   EXPECT_CALL(client_factory, Call).Times(0);
   auto creds =
       GoogleDefaultCredentials(Options{}, client_factory.AsStdFunction());
+  (void)std::remove(filename.c_str());
   ASSERT_STATUS_OK(creds);
-  auto* ptr = creds->get();
-  EXPECT_EQ(typeid(*ptr), typeid(AuthorizedUserCredentials));
+  EXPECT_THAT(creds->get(),
+              WhenDynamicCastTo<AuthorizedUserCredentials*>(NotNull()));
+}
+
+TEST_F(GoogleCredentialsTest, LoadValidExternalAccountCredentialsViaEnvVar) {
+  auto const filename = TempFileName();
+  std::ofstream(filename) << kExternalAccountContents;
+  auto const env = ScopedEnvironment(GoogleAdcEnvVar(), filename.c_str());
+
+  // Test that the authorized user credentials are loaded as the default when
+  // specified via the well known environment variable.
```

Comment:
Nit: "well known" should be hyphenated when it modifies a noun.

Reply:
Fixed here and elsewhere on the file